src/diffusers/models/attention_processor.py

```diff
@@ -692,6 +694,28 @@ def norm_encoder_hidden_states(self, encoder_hidden_states: torch.Tensor) -> tor
 
         return encoder_hidden_states
 
+    @torch.no_grad()
+    def enable_fused_projections(self, device, dtype, fuse_projections=True, is_cross_attention=False):
```

Comment:
```suggestion
    def enable_fused_projections(self, device, dtype, fuse_projections=True):
```

Can't we retrieve whether we're in cross attention mode just from: `self.cross_attention_dim == self.query_dim` (we probably need to save query_dim in init)

Also why do we need to pass device and dtype? Can't we just retrieve it from the prev projection layers?